### ISSUE TITLE
[Dynamic Form] Always Show Required Field Validation Error In "FormDisplayMode.Edit" Mode

### ISSUE THREAD

**Issue by wuxiaojun514** [pnp/sp-dev-fx-controls-react#1775]
#### Category
[ ] Enhancement

[x] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [ 3.17  ]. It works well in [3.16] 
Dynamic Form refactored a lot in 3.17 

#### Expected / Desired Behavior / Question
Ideally when you edit an existing item with "Dynamic Form", it should not show "You can't leave this blank." error if this field already has data.

#### Observed Behavior
When in Edit mode (PageType=6), Dynamic Form control will always show "You can't leave this blank." error on required field even it already has existing data on current item. 
![image](https://github.com/pnp/sp-dev-fx-controls-react/assets/29603048/f1758253-8b4d-42f9-83f2-b0596e6e78df)

#### Steps to Reproduce
1. Choose a list which has a required field (e.g. title) and create a new item. Make sure this new created item has data on required field.
2. Put "Dynamic Form" in the spfx form customization, no special settings.

```
     <DynamicForm
          context={this.props.context}
          listId={this.props.context.list.guid.toString()}
          listItemId={this.props.context.itemId}
          onSubmitted={(listItemData: any) => {
            console.log(listItemData);       
              setTimeout(() => {
                this.props.onSave();
              }, 1000);
          }} 
          onListItemLoaded={async (listItemData: any) => { // eslint-disable-line @typescript-eslint/no-explicit-any
            console.log(listItemData);
          }} />
```
3.  Update EditForm Config in "serve.json" to point to this new created item, then launch it with edit form config
![image](https://github.com/pnp/sp-dev-fx-controls-react/assets/29603048/2e99a319-d462-49a4-a2a4-19c2ea805f00)

4. You will see "You can't leave this blank." error on this item even it already has data on this field
![image](https://github.com/pnp/sp-dev-fx-controls-react/assets/29603048/6f087020-87a2-4b82-adb6-d2bead411497)

**Comment by NishkalankBezawada** [comment]
Hello, 

We are also facing the same error with version 3.18.1. @wuxiaojun514 Any update on the fix?

Thanks,
Nishkalank

**Comment by wuxiaojun514** [comment]
Hi @NishkalankBezawada  I created a PR [1781](https://github.com/pnp/sp-dev-fx-controls-react/pull/1781) for it. But it seems waiting some other PR [1726](https://github.com/pnp/sp-dev-fx-controls-react/pull/1726) getting approval first. Right now my fix has not been merged into Dev branch yet

**Comment by RobertSiE** [comment]
Hi, do you have any news regarding to this bug?

**Comment by wuxiaojun514** [comment]
Hi @RobertSiE. I don't have any update on it.  I saw the author of #1726 seems inactive for quite a long time, and several other PRs are also waiting for this pr to be resolved first.

**Comment by hena-s** [comment]
Any update on this issue

**Comment by gchamplin** [comment]
This is still a problem and it's been several months without any progress.

**Comment by gchamplin** [comment]
I see it when I set the ContentTypeId in the Dynamic form. If I leave that prop out I don't see the error message.

**Comment by joelfmrodrigues** [comment]
Hey all, the fix for this has finally been merged, apologies for the long delay

**Comment by gchamplin** [comment]
How do I migrate an existing project using v 3.19.0 to use the fixed version (3.20.0). 
I tried doing an npm install again with --@3.20.0 but it didn't change in node modules. Do I need to uninstall 3.19.0 first?

**Comment by wuxiaojun514** [comment]
Hi @gchamplin 
Based my experience , it would be better uninstall the old version and install new version again.
